### PR TITLE
add feature register and login

### PR DESCRIPTION
**Pull request by xrzky**

### CODE REVIEW

File: config/config.js
Comment (hanifwdyt) [inline config/config.js:1]: naming config/database.js or config/db.js

**Review by hanifwdyt — CHANGES_REQUESTED** [review]

File: models/user.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+    static associate(models) {
+      // define association here
+    }
+  }
+  User.init({
```
Comment (hanifwdyt) [inline models/user.js:17]: is this table users does not have user id?

File: middlewares/error-middleware.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+function errorMiddleware(error, req, res, next) {
+    let code = 500;
+    let message = 'Internal Server Error';
+
+    if (error.name === 'SequelizeUniqueConstraintError') {
+        res.status(400).json({ message: 'Bad Request' });
```
Comment (hanifwdyt) [inline middlewares/error-middleware.js:6]: in class tomorrow please give all of this example case for error handling

**Review by hanifwdyt — CHANGES_REQUESTED** [review]

File: config/config.js
Comment (xrzky) [reply config/config.js:1]: okay sir

File: models/user.js (hunk 1 above)
Comment (xrzky) [reply models/user.js:17]: The user_id is located in the migration file where I create a database table using Sequelize.

File: middlewares/error-middleware.js (hunk 2 above)
Comment (xrzky) [reply middlewares/error-middleware.js:6]: noted sir

**Pull request merged**